client/my-sites/plugins/plugins-browser-item/index.jsx

```diff
@@ -271,11 +285,16 @@ const InstalledInOrPricing = ( {
 								<>
 									{ price + ' ' }
 									<span className="plugins-browser-item__period">{ period }</span>
+									{ shouldUpgradeToInstallPurchasedPlugins && (
+										<div className="plugins-browser-item__period">
+											{ translate( 'Requires a plan upgrade' ) }
+										</div>
+									) }
 								</>
 							) : (
 								<>
 									{ translate( 'Free' ) }
-									{ shouldUpgrade && (
+									{ shouldUpgradeToManagePlugins && (
```

Comment:
Rather than changing `shouldUpgradeCheck()` and everything that calls it, could we define a `hasInstallPlugin` const at the top of this component and add a check for it here? Would that work?
```suggestion
									{ shouldUpgrade && ! hasInstallPlugins && (
```

Reply:
🤔 There is an issue that `shouldUpgradeCheck` in trunk checks for `WPCOM_FEATURES_INSTALL_PURCHASED_PLUGINS`. So a check for "free plugins" access with `shouldUpgrade && ! hasInstallPlugins` would return false on a Starter plan, because Starter plans return false for `shouldUpgrade`.

We could do `shouldUpgrade || ! hasInstallPlugins`, or just `! hasInstallPlugins`, but we have to consider Jetpack sites. `shouldUpgradeCheck` is already doing the feature check and considering Jetpack, is why I co-opted it. I figured it'd be OK, since it is not widely used.

Reply:
I feel like adding a parameter to `shouldUpgradeCheck()` and building layers on top of `shouldUpgrade()` is making things too complex (and extending the use of a function with an ambiguous name).

Instead of building upon `shouldUpgrade()`, one thought I had was breaking it down:
(rough idea)

```js
const siteIsMarketplaceAddonCompatible = ( state: IAppState, siteId: number | null ): boolean | null => {
	if ( ! siteId ) {
		return null;
	}
	const isStandaloneJetpack =
		isJetpackSite( state, siteId ) && ! isSiteAutomatedTransfer( state, siteId );
	const isVip = isVipSite( state, siteId );
	return ! isStandaloneJetpack && ! isVip;
};
```

and directly using checks like `abc: siteIsMarketplaceAddonCompatible( state, siteId ) && siteHasFeature( state, siteId, WPCOM_FEATURES_INSTALL_PLUGINS )`. I think it might be easier to follow what it's doing, any thoughts?

Reply:
That seems like a good approach, @mreishus. I assume we'd remove `shouldUpgradeCheck` and replace its usage with this and a feature check?

Reply:
Yeah, I think either way could work (new stuff only vs. updating previous)